File: leetcode/2097-watering-plants/wateringPlants.test.ts
import cases from "./cases.json";
import wateringPlants from "./wateringPlants";

test("watering plants", () => {
  cases.forEach((testCase, index) => {
    console.log(`正在执行第 ${index + 1} 个测试用例\n`);
    const result = wateringPlants(
      testCase.input.plants,
      testCase.input.capacity
    );
    console.log(
      `input: ${testCase.input} , output: ${testCase.output} ==> ${result} \n`
    );

    expect(result).toBe(testCase.output);
  });
});

File: leetcode/2097-watering-plants/wateringPlants_test.rs
use std::{env::current_dir, path::Path};

use serde::Deserialize;

use crate::{util::load_cases, Solution};

#[derive(Deserialize, Debug)]
struct Input {
    plants: Vec<i32>,
    capacity: i32,
}

#[derive(Deserialize, Debug)]
struct Case {
    input: Input,
    output: i32,
}

#[test]
fn test_watering_plants() {
    let current_path = current_dir().unwrap().join(file!());
    let dir = current_path.parent().unwrap();

    let cases: Vec<Case> = load_cases(Path::new(&dir.join("./cases.json")));

    for case in &cases {
        assert_eq!(
            Solution::watering_plants(case.input.plants.clone(), case.input.capacity),
            case.output
        );
    }
}
